ocs_ci/ocs/ui/base_ui.py

```diff
@@ -0,0 +1,178 @@
+import time
+
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as ec
+from selenium.webdriver.common.by import By
+from webdriver_manager.chrome import ChromeDriverManager
+from webdriver_manager.utils import ChromeType
+
+from ocs_ci.framework import config as ocsci_config
+from ocs_ci.utility.utils import run_cmd, get_kubeadmin_password
+from ocs_ci.ocs.ui.views import login
+
+
+logger = logging.getLogger(__name__)
+
+
+class BaseUI:
+    """
+    Base Class for UI Tests
+
+    """
+
+    def __init__(self, driver):
+        self.driver = driver
+        self.screenshots_folder = os.path.join(
+            os.path.expanduser(ocsci_config.RUN["log_dir"]),
+            f"screenshots_ui_{ocsci_config.RUN['run_id']}",
+        )
+        os.mkdir(self.screenshots_folder)
+        logger.info(f"screenshots pictures:{self.screenshots_folder}")
+        self.cnt_screenshot = 0
+
+    def do_click(self, by_locator, type=By.XPATH, timeout=30):
+        """
+        Click on Button/link on OpenShift Console
+
+        by_locator (str): GUI element needs to operate on.
+        type (By): Set of supported locator strategies.
+        timeout (int): Looks for a web element repeatedly until timeout (sec) happens.
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        element.click()
+        screenshot = ocsci_config.UI_SELENIUM.get("screenshot")
+        if screenshot:
+            self.take_screenshot()
+
+    def do_send_keys(self, by_locator, text, type=By.XPATH, timeout=30):
+        """
+        Send text to element on OpenShift Console
+
+        by_locator (str): GUI element needs to operate on.
+        text (str): Send text to element
+        type (By): Set of supported locator strategies
+        timeout (int): Looks for a web element repeatedly until timeout (sec) happens.
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        element.send_keys(text)
+
+    def is_expanded(self, by_locator, type=By.XPATH, timeout=30):
+        """
+        Check whether an element is in an expanded or collapsed state
+
+        Args:
+            by_locator (str): GUI element needs to operate on.
+            type (By): Set of supported locator strategies
+            timeout (int): Looks for a web element repeatedly until timeout (sec) happens.
+
+        return:
+            bool: True if element expended, False otherwise
+
+        """
+        wait = WebDriverWait(self.driver, timeout)
+        element = wait.until(ec.element_to_be_clickable((type, by_locator)))
+        return True if element.get_attribute("aria-expanded") == "true" else False
+
+    def choose_expanded_mode(self, mode, by_locator, type):
```

Comment:
Just cosmetic suggestion - I would reorder the parameters to have them in the same/similar order as in other methods - if applicable (same parameters on the same positions). For example in the previous two methods, `by_locator` is the second parameter, here it is third.